SalesforceCommerceCloud/pwa-kit
# @W-14499317: Treat plus sign as space in query string

# Description

In a URL, the plus sign (+) is a valid representation of a space. However, PWA does not handle + characters as expected when + characters are included as part of the search query string. We should update PWA so that it is able to handle + (and possibly other special characters; Newell has reported degree ° and ampersand &) as expected when they are included in the search query string.

Problem:
- We have the function `getLocationSearch` in react-rendering where we process url to be able to handle multi-val, during the process, we append the key-value from the req.query to `newParams` object by using URLSearchParams.append. While URLSearchParams constructor treats plus sign (+) as space when you first initialized, its append method does not, and we want to retain that behavior during appending.

Solution:
- We run the key-val through URLSearchParams before appending into the newParams objects



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out code
- npm ci
- run Retail App
- Go to this page [http://localhost:3000/global/en-GB/search?limit=25&q=men+shirt&refine=price%3D%2820..50%29&refine=c_size%3D16&refine=c_refinementColor%3DBlue%7CWhite&sort=best-matches](http://localhost:3000/global/en-GB/search?limit=25&q=men+shirt&refine=price%3D%2820..50%29&refine=c_size%3D16&refine=c_refinementColor%3DBlue%7CWhite&sort=best-matches)
- Observe that no error is thrown
- + in the url is treated as space for search query

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
